### PR TITLE
Delete Conversation Reference from ConversationIdFactory when SkillDialog receives EndOfConversation while using ExpectReplies

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes: https://github.com/microsoft/botbuilder-dotnet/issues/4567

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/SkillDialogTests.cs [hunk 1]
```diff
@@ -438,7 +478,9 @@ private Activity CreateSendActivity()
         // Simple conversation ID factory for testing.
         private class SimpleConversationIdFactory : SkillConversationIdFactoryBase
         {
-            private readonly ConcurrentDictionary<string, SkillConversationReference> _conversationRefs = new ConcurrentDictionary<string, SkillConversationReference>();
+#pragma warning disable SA1401 // Fields should be private
+            internal readonly ConcurrentDictionary<string, SkillConversationReference> ConversationRefs = new ConcurrentDictionary<string, SkillConversationReference>();
```
Comment (gabog) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/SkillDialogTests.cs:482]: Just make it an auto property and remove the pragma, it is easier to read.

**Review by gabog — CHANGES_REQUESTED** [review]
Just one small correction and we are good. 
Thanks

**Review by gabog — APPROVED** [review]

**Comment by gabog** [comment]
@EricDahlvang , the build seems to be failing, can you take a look?
Thanks

**Comment by EricDahlvang** [comment]
Build is passing now.  But, apparently this requires approval from one of: https://github.com/orgs/microsoft/teams/bb-dotnet/members

**Comment by gabog** [comment]
@mrivera-ms  / @cleemullins, can you review and approve this one?

Thanks

**Review by mrivera-ms — APPROVED** [review]

**Pull request merged**